secrets/secrets-vault/src/lib.rs

```diff
@@ -0,0 +1,497 @@
+}
+
+#[derive(Serialize, Deserialize)]
+struct VaultAuthClaims {
+    aud: String,
+    iss: String,
+    sub: String,
+    exp: u64,
+    application: String,
+    host: Claims<Host>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    component: Option<Claims<Component>>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    provider: Option<Claims<CapabilityProvider>>,
+}
+
+struct VaultSecretRef {
+    secret_path: String,
+    version: Option<u64>,
+}
+
+impl TryFrom<&SecretRequest> for VaultSecretRef {
+    type Error = anyhow::Error;
+
+    fn try_from(request: &SecretRequest) -> StdResult<Self, Self::Error> {
+        let secret_path = request.name.clone();
+        let version = if let Some(version) = request.version.clone() {
+            version
+                .parse::<u64>()
+                .map(Some)
+                .map_err(|_| anyhow!("unable to convert requested version to integer"))?
+        } else {
+            None
+        };
+
+        Ok(Self {
+            secret_path,
+            version,
+        })
+    }
+}
+
+pub struct SubjectMapper {
+    pub prefix: String,
+    pub service_name: String,
+}
+
+impl SubjectMapper {
+    pub fn new(prefix: &str, service_name: &str) -> Result<Self> {
+        Ok(Self {
+            prefix: prefix.to_string(),
+            service_name: service_name.to_string(),
+        })
+    }
+
+    fn queue_group_name(&self) -> String {
+        format!("{}.{}", self.prefix, self.service_name)
+    }
+
+    fn secrets_subject(&self) -> String {
+        format!(
+            "{}.{}.{}",
+            self.prefix, SECRETS_API_VERSION, self.service_name
+        )
+    }
+
+    fn secrets_wildcard_subject(&self) -> String {
+        format!("{}.>", self.secrets_subject())
+    }
+}
+
+pub struct VaultSecretsBackend {
+    nats_client: async_nats::Client,
+    nkey: nkeys::KeyPair,
+    xkey: nkeys::XKey,
+    jwks_address: SocketAddrV4,
+    subject_mapper: SubjectMapper,
+    vault_config: VaultConfig,
+}
+
+pub struct VaultConfig {
+    pub address: String,
+    pub auth_mount: String,
+    pub audience: String,
+}
+
+impl VaultSecretsBackend {
+    pub fn new(
+        nats_client: async_nats::Client,
+        nkey: KeyPair,
+        xkey: XKey,
+        jwks_address: SocketAddrV4,
+        subject_mapper: SubjectMapper,
+        vault_config: VaultConfig,
+    ) -> Self {
+        Self {
+            nats_client,
+            nkey,
+            xkey,
+            jwks_address,
+            subject_mapper,
+            vault_config,
+        }
+    }
+
+    pub async fn serve(&self) -> anyhow::Result<()> {
+        let pk = KeyPair::from_public_key(&self.nkey.public_key())?;
+        tokio::spawn({
+            let listen_address = self.jwks_address.to_owned();
+            async move {
+                VaultSecretsJwksServer::new(vec![pk], listen_address)
+                    .unwrap()
+                    .serve()
+                    .await
+                    .unwrap()
```

Comment:
Should this never fail, or could we at least replace these `unwrap`s with `expect`s to provide some message?